docs/operations.md

```diff
@@ -18,10 +18,54 @@ then deleting the old chunks. The scavenged events are also removed from the ind
 
 Running a scavenge causes the active chunk to be completed so that it can be scavenged.
 
-::: warning 
+::: warning
 Scavenging is destructive. Once a scavenge has run, you cannot recover any deleted events except from a backup.
 :::
 
+### Mechanism of the scavenging
+
+Running scavenge for the first time will take longer compared to subsequent scavenges because it needs to create a "scavenge point" as part of the first phase, known as the accumulation phase. This is followed by two more phases at the end of which events are removed from the database, including from the current chunk. More information on these phases can be found below. 
+
+Whenever a scavenging process starts on a node, it will mark the current active chunk as closed and checks if there are any existing unused scavenge point on that. If there are, it utilizes one of them; otherwise, it creates a new scavenge point and replicates it to another node.
+
+_Scavenge point always get replicated to all other nodes, allowing you to run scavenging up to the same scavenge point on other nodes using the SyncOnly parameter._
+
+#### Scavenge point
+
+The scavenge point includes the following information:
+
+"Position": the position at which the scavenge will run up to.
+
+"EventNumber": the number assigned to the scavenge point.
+
+"EffectiveNow": the time used to calculate the maxAge of an event (the time when the scavenge point was written).
+
+"Threshold": the minimum weight that a chunk must have before it is executed.
+
+**The scavenging process consists of three phases:**
+
+#### 1. Accumulation
+
+During this phase, the scavenging process finds the latest scavenge point (if it doesn't exist, it creates one) and reads through the log from the beginning up to that point, it accumulates necessary information for the scavenge process (such as maxAge, maxCount,stream name, min\max event's create time, etc) into the scavenge database.
+
+_A chunk only need to be accumulated once._
+
+#### 2. Calculation
+
+In this phase, the scavenging process reads through the index and calculates which events can be discarded and in which chunks they are located in. It assigns weight to those chunks.
+
+#### 3. Execution
+
+The final phase involves removing all the events and index that need to be scavenged base on the data obtained from the calculation phase. Additionally, small chunks are merged together.
+
+This phase includes the following sub-phases:
+* Chunk Execution: removal of all related chunks.
+* Chunk merging: merging of small chunks.
+* Index execution: removal of all related indexes.
+* Cleaning: cleaning of checkpoint and other data.
```

Comment:
cleaning is its own phase really. it removes data from the scavenge database that is no longer needed